projects/jsf-validation/src/lib/schema-validation.service.ts

```diff
@@ -0,0 +1,88 @@
+import Ajv, { ErrorObject } from 'ajv';
+import { isNil } from 'lodash-es';
+import { JSFJsonSchema } from '../../../jsf/src/lib/jsf-json-schema';
+
+export class SchemaValidationService {
+  private static properties = 'properties';
+  private static missingProperty = 'missingProperty';
+  private static nameProperty = 'name';
+  private static isHidden = 'isHidden';
+  private static isReadOnly = 'isReadOnly';
+
+  /**
+   * Validate the JSON data against the schema using AJV
+   * @param schema: schema to validate against
+   * @param values: values to validate
+   */
+  public static validate(schema: JSFJsonSchema, values: any): JSFErrorObject[] {
+    const ajv = new Ajv({allErrors: true});
+    const valid = ajv.validate(schema, values);
+    if (!valid) {
+      const errors = ajv.errors.map(error => SchemaValidationService.toJsfError(ajv, error));
+      return errors.filter(error => !error.errorSchema[SchemaValidationService.isHidden])
+        .filter(error => !error.errorSchema[SchemaValidationService.isReadOnly]);
+    }
+    return null;
+  }
+
+  private static toJsfError(ajv: Ajv.Ajv, error: ErrorObject): JSFErrorObject {

```

Comment:
Is this validation scoped to your story and the schema you're creating for the S3 connector? 
I'm wondering if this will fail for some of our more complex schema or not (Like what is in `schemaV2.json`)

Reply:
Scoped to the S3 connector for now. The schemaV2.json will have a false-negative for nested required fields etc. We can discuss more this afternoon

Reply:
Ok. If this is the case, I think we should make it very clear that this validator is not ready for all use cases. Probably just need to add some comments at the top of the file clarifying this.